app/services/open_library_service.rb

```diff
@@ -0,0 +1,11 @@
+class OpenLibraryService
+  include HTTParty
+  base_uri 'https://openlibrary.org/api/books'
+
+  def self.execute(isbn)
+    response = get('/', query: { bibkeys: "ISBN:#{isbn}", jscmd: 'data', format: 'json' })
+    raise ActiveRecord::RecordNotFound, 'Book not found' if response.parsed_response.empty?
```

Comment:
Mm i don't know if RecordNotFound is a good exception, this has nothing to do with the db, RuntimeError could be a good one